applications/GeoMechanicsApplication/tests/cpp_tests/custom_constitutive/test_interface_coulomb_with_tension_cut_off.cpp

```diff
@@ -0,0 +1,456 @@
+#include "geo_mechanics_application_variables.h"
+#include "includes/stream_serializer.h"
+#include "tests/cpp_tests/geo_mechanics_fast_suite.h"
+#include "tests/cpp_tests/test_utilities.h"
+
+#include <boost/numeric/ublas/assignment.hpp>
+
+using namespace Kratos;
+using namespace std::string_literals;
+
+namespace
+{
+
+using namespace Kratos;
+
+void InitializeLawMaterial(ConstitutiveLaw& rLaw, const Properties& rProperties)
+{
+    const auto dummy_element_geometry      = Geometry<Node>{};
+    const auto dummy_shape_function_values = Vector{};
+    rLaw.InitializeMaterial(rProperties, dummy_element_geometry, dummy_shape_function_values);
+}
+
+} // namespace
+
+namespace Kratos::Testing
+{
+
+KRATOS_TEST_CASE_IN_SUITE(InterfaceCoulombWithTensionCutOff_Clone, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    // Arrange
+    const auto original_law = InterfaceCoulombWithTensionCutOff{};
+
+    // Act
+    auto p_cloned_law = original_law.Clone();
+
+    // Assert
+    KRATOS_EXPECT_NE(p_cloned_law.get(), nullptr);
+    KRATOS_EXPECT_NE(p_cloned_law.get(), &original_law);
+    KRATOS_EXPECT_NE(dynamic_cast<const InterfaceCoulombWithTensionCutOff*>(p_cloned_law.get()), nullptr);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(InterfaceCoulombWithTensionCutOff_CalculateMaterialResponseCauchyAtElasticZone,
+                          KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    // Arrange: set elastic compressive state
+    auto properties = Properties{};
+    properties.SetValue(GEO_FRICTION_ANGLE, 35.0);
+    properties.SetValue(GEO_COHESION, 10.0);
+    properties.SetValue(GEO_DILATANCY_ANGLE, 20.0);
+    properties.SetValue(GEO_TENSILE_STRENGTH, 10.0);
+
+    auto parameters = ConstitutiveLaw::Parameters{};
+    parameters.SetMaterialProperties(properties);
+    auto traction_vector = Vector{ZeroVector{2}};
+    parameters.SetStressVector(traction_vector);
+    auto relative_displacement_vector = Vector{ZeroVector{2}};
+    parameters.SetStrainVector(relative_displacement_vector);
+
+    auto p_initial_state         = make_intrusive<InitialState>();
+    auto initial_traction_vector = Vector{2};
+    initial_traction_vector <<= -6.0, 8.0;
+    p_initial_state->SetInitialStressVector(initial_traction_vector);
+    p_initial_state->SetInitialStrainVector(Vector{ZeroVector{2}});
+
+    auto law = InterfaceCoulombWithTensionCutOff{};
+    law.SetInitialState(p_initial_state);
+    InitializeLawMaterial(law, properties);
+
+    // Act
+    law.CalculateMaterialResponseCauchy(parameters);
+
+    // Assert
+    auto expected_traction_vector = Vector{2};
+    expected_traction_vector <<= -6.0, 8.0;
+    KRATOS_EXPECT_VECTOR_NEAR(parameters.GetStressVector(), expected_traction_vector, Defaults::absolute_tolerance);
+
+    // Arrange: set elastic tensile state
+    traction_vector <<= 5.0, 4.0;
+    const auto dummy_process_info = ProcessInfo{};
+    law.SetValue(CAUCHY_STRESS_VECTOR, traction_vector, dummy_process_info);
```

Comment:
We can get rid of a local variable by using a temporary object:
```suggestion
    law.SetValue(CAUCHY_STRESS_VECTOR, traction_vector, ProcessInfo{});
```
In terms of readability, in my opinion, both ways are similar. The suggestion is, however, simpler.